### PR TITLE
[EMCAL-539,EMCAL-696,EMCAL-697] FIT-CTP-EMCAL trigger simulation

### PR DESCRIPTION
**Pull request by siragoni**
- added the EMCAL trigger simulation in the Digitizer workflow
- added the CTP input digits with the result of the trigger simulation
- fixed MB trigger string in CTP simulation
- added the CTP-EMC L0 accept

### CODE REVIEW

**Comment by siragoni** [comment]
@mfasDa @jokonig Hi Markus, Joshua, I think I am ready now for you to have a look and point out edits you would like to be implemented

**Review by jokonig — DISMISSED** [review]

File: Detectors/EMCAL/simulation/src/DigitizerTRU.cxx [hunk 1]
```diff
@@ -436,4 +437,77 @@ void DigitizerTRU::finish()
...
+    return id;
+  } else if (detector == kDCAL) {
+    return 32 + ((int)(id / 4) * 6) + ((id % 4 < 2) ? (id % 4) : (id % 4 + 2));
+  }
+  return -1;
+}
+//______________________________________________________________________
+int DigitizerTRU::GetChannelForMaskRun2(int mask, int bitnumber, bool onethirdsm)
+{
+  if (onethirdsm)
```
Comment (jokonig) [inline Detectors/EMCAL/simulation/src/DigitizerTRU.cxx]: This has to be in brackets

Comment (siragoni) [reply Detectors/EMCAL/simulation/src/DigitizerTRU.cxx]: Done

**Review by jokonig — APPROVED** [review]

**Comment by ktf** [comment]
Adding @sawenzel @shahor02 for the review.

File: Detectors/EMCAL/simulation/include/EMCALSimulation/LZEROElectronics.h [hunk 2]
```diff
@@ -76,6 +78,10 @@ class LZEROElectronics
   /// Sets geometry for trigger mapping
   void setGeometry(o2::emcal::Geometry* gm) { mGeometry = gm; }
 
+  /// Sets the masked fastOrs from the CCDB in the LZERO
+  void setMaskedFastOrs(std::vector<int> maskedfastors) { mMaskedFastOrs = maskedfastors; }
```
Comment (shahor02) [inline Detectors/EMCAL/simulation/include/EMCALSimulation/LZEROElectronics.h:82]: pass `maskedfastors` as `const std::vector<int>& maskedfastors`

**Review by shahor02 — COMMENTED** [review]
just one technical comment but I did not follow the changes in logic.

File: Detectors/EMCAL/simulation/include/EMCALSimulation/LZEROElectronics.h (hunk 2 above)
Comment (siragoni) [reply Detectors/EMCAL/simulation/include/EMCALSimulation/LZEROElectronics.h:82]: Hi @shahor02 , does it mean that once I implement this change you are able to approve this PR and merge it, so you have the missing rights @jokonig and I need?

**Comment by shahor02** [comment]
Yes, technically, I can merge it, but it would be good if @sawenzel approved it since this is a simulation code.
I will add @jokonig as a code owner instead of @mfasDa .

**Review by sawenzel — APPROVED** [review]

**Comment by sawenzel** [comment]
Should squash the commits together to keep as an atomic development.

**Comment by shahor02** [comment]
I've checked, actually, @jokonig is the code owner of the EMCAL directories, but here the modifications are in the CTP ...

**Comment by siragoni** [comment]
> I've checked, actually, @jokonig is the code owner of the EMCAL directories, but here the modifications are in the CTP ...

Hi @shahor02 yes, this is why I was saying we have missing rights. There were a few modifications that we did over the course of months, a few of them Roman implemented himself afterwards, but still one thing that Markus wanted to add are here in this PR

**Pull request merged**